requirement-generator/types.ts

```diff
@@ -45,7 +45,7 @@ export interface CollegeOrMajorRequirement extends BaseRequirement {
 }
 
 export interface DecoratedCollegeOrMajorRequirement extends BaseRequirement {
-  readonly courses: readonly string[];
+  readonly courses: 'all-eligible' | readonly string[];
```

Comment:
Is there a reason that courses could either be a string or list? What if it's ['all-eligible']?

Reply:
It won't be. The case of string list is literally a list of all courses, like
```
["CS2112", "CS2800"]
```

Reply:
Right, but it could also be 'all-eligible'?

Reply:
The `getSatisfiableCourses` is the function that generates the array. It will never put `all-eligible` in that array,